pkgs/applications/misc/scli/default.nix

```
@@ -0,0 +1,24 @@
+{ lib, buildPythonApplication, fetchFromGitHub, signal-cli, urwid, urwid-readline, dbus }: buildPythonApplication rec {
+  pname = "scli";
+  version = "0.6.0";
+  src = fetchFromGitHub {
+    owner = "isamert";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1q2p0i4ar5xlpj4nbhadj67k9r7m7ph2x15q8zlw9bch6lixv8rg";
+  };
+  propagatedBuildInputs = [ signal-cli urwid urwid-readline dbus ];
+  dontBuild = true;
+  doCheck = false;
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m755 -D scli $out/bin/scli
```

```suggestion
    install -m755 -D scli $out/bin/scli
    wrapProgram $out/bin/scli --prefix PATH : ${lib.makeBinPath [ signal-cli ]}
```
This should get scli to be able to pickup signal-cli on runtime.
More info on makeWrapper/wrapProgram: https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-functions

---

Would this make the binary signal-cli available to the user or just to the scli program itself? The issue that I've run into is that scli expects the user to have access to signal-cli to perform certain auth tasks, e.g.
```
$ signal-cli -u USERNAME register
```
now I think the program will work fine with the current implementation if the user already has a signal account, but I'm worried it might be confusing for the user if they have to manually install both? I don't know what precedent is here.

---

I'm thinking about it, and as a user I would probably understand and not think too much of it that I would have to explicitly install signal-cli if I wanted to run signal-cli directly myself. I think the current state of the package is probably acceptable?

---

It is possible to symlink signal-cli's binary to `scli`'s out path. I'm not sure if that would be acceptable by nixpkgs standards. But it does make sense that the user would realize that signal-cli needs to be installed separately for registration. So I think this implementation is fine.

---

Did you end up addressing the issue of `signal-ci` not being in the path of scli? It looks like `install` is still being used.